app/models/invoice/data_presenter.rb

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+
+class Invoice
+  class DataPresenter
+    attr_reader :invoice
+
+    delegate :data, :date, to: :invoice
+
+    FINALIZED_NON_SUCCESSFUL_STATES = %w(canceled returned).freeze
+
+    extend Invoice::DataPresenterAttributes
+
+    attributes :additional_tax_total, :currency, :included_tax_total, :payment_total,
+               :shipping_method_id, :state, :total
+    attributes :number, :note, :special_instructions, prefix: :order
```

Comment:
Is there a reason for prefixing these three order attributes? 

I think it would be better if a regular `Spree::Order` object and an instance of a `Invoice::DataPresenter` data object have the same attribute names. That way they could be used interchangeably in more situations, like [Duck Types](https://en.wikipedia.org/wiki/Duck_typing).

```ruby
real_order_object = Spree::Order.find(23)
order_from_invoice = Invoice::DataPresenter.new(order.invoices.first)

real_order_object.number # => 23
order_from_invoice.number # => 23
```

Currently `order_from_invoice.number` would throw a `MethodMissing` error but `order_from_invoice.order_number` would return `23`. Know what I mean? I think there will be benefits later on if they both have the same interface.

So... that would mean removing the `prefix: :order` for these three definitions and, and then changing all instances here of `.order_number`, `.order_note`, and `.order_special_instructions` to `.note`, `.number`, and `.special_instructions`. And then if you've got invoices saved in your development database you would have to delete them with `Invoice.destroy_all` and remake them because the format will be changed.

One benefit of this change is that we would be able to render an invoice object completely independently of the `order` object, like:
```ruby
@order = Invoice.find(params[:invoice_id]).presenter

InvoiceRenderer.new.args(@order)
```

What do you think?

Reply:
That was helpful.
Thank you @Matt-Yorkley  :pray: